app/components/ui/TokenLogo.tsx

```diff
@@ -0,0 +1,64 @@
+'use client';
+
+import { Logo, LogoSizeProp } from '@swapr/ui';
+import { useQueryClient } from '@tanstack/react-query';
+import { useEffect, useState } from 'react';
+import { WXDAI } from '@/constants';
+import { XDAI_LOGO, DEFAULT_TOKEN_LOGO } from '@/public/assets';
+
+const localTokenAssets = [{ address: WXDAI.address, asset: XDAI_LOGO.src }];
+
+const fetchImage = async (url: string) => {
+  const result = await fetch(url);
+
+  if (result.status === 200) return url;
+  else throw null;
```

Comment:
Mmm, can we handle this error more descriptively? It feels weird to throw `null` on error.

Reply:
The thing is, nothing is done with this error.  But will add a message just in case.